mm2src/coins/lp_coins.rs

```diff
@@ -276,7 +313,35 @@ impl Deref for TransactionEnum {
     }
 }
 
-pub type TransactionFut = Box<dyn Future<Item = TransactionEnum, Error = String> + Send>;
+#[derive(Debug, Clone)]
+pub enum TransactionErr {
+    /// Keeps transactions while throwing errors.
+    TxRecoverableError(Box<TransactionEnum>, String),
```

Comment:
Did Clippy require `Box` here?

Reply:
Yes it did.

Reply:
Sure, but I am curious what is the specific reason for this?

Reply:
> https://rust-lang.github.io/rust-clippy/master/#large_enum_variant - more info on the lint slightly_smiling_face

I see now, thank you :)